server/src/main/java/org/elasticsearch/cluster/coordination/NodeJoinExecutor.java

```diff
@@ -123,7 +124,17 @@ public ClusterState execute(BatchExecutionContext<JoinTask> batchExecutionContex
             newState = ClusterState.builder(initialState);
         } else {
             logger.trace("processing node joins, but we are not the master. current master: {}", currentNodes.getMasterNode());
-            throw new NotMasterException("Node [" + currentNodes.getLocalNode() + "] not master for join request");
+            throw new NotMasterException(
+                "Node ["
+                    + currentNodes.getLocalNode()
+                    + "] not master for join request. Current known master: ["
+                    + currentNodes.getMasterNode()
+                    + "], current term: ["
+                    + term
+                    + "], current local time: [ "
+                    + TimeValue.timeValueMillis(System.currentTimeMillis())
```

Comment:
This yields a string something like `19852.6d` which I don't think is what you intended? Are you sure the time is even useful here? We report the age of the exception in the `JoinHelper` message IIRC.

Reply:
Hmm. It doesn't look like we normally convert times to human readable dates: no standard way to do it. I'll throw out this bit, thanks for pointing that out.

I wanted to know the time frame in which this node returned NotMaster. On the requesting node, we get back a NotMaster error, but then I want to be able to go to the NotMaster node's logs and see what was happening at that time. It's not strictly necessary, the requesting node happens to track the time at which the error response was received and will log how many milliseconds ago that was, but having a specific time would still be a nice shortcut.